src/main/java/com/microsoft/azure/proton/transport/proxy/impl/ProxyHandlerImpl.java

```diff
@@ -56,9 +56,7 @@ public ProxyResponseResult validateProxyResponse(ByteBuffer buffer) {
             final Scanner responseScanner = new Scanner(response);
             if (responseScanner.hasNextLine()) {
                 final String firstLine = responseScanner.nextLine();
-                if ((firstLine.toLowerCase().contains("http/1.1") || firstLine.toLowerCase().contains("http/1.0"))
-                        && firstLine.contains("200")
-                        && firstLine.toLowerCase().contains("connection established")) {
+                if (firstLine.toLowerCase().trim().matches("^http\\/(1.0|1.1)\\s(200|201|202|203|204|205|206)\\s\\w.*$")) {

```

Comment:
* I think using a member variable `Pattern` would be easier because we wouldn't have to recompile it every time. 
* `\s` would compare any whitespace, but the spec says only the SP character. From what I read, CR|LF is only valid at the end of the line.
* Since we don't care about the "Status description" we can stop matching after the status code, so we don't have to use `\w.$`.


```java
class ProxyHandlerImpl {
    // Indicates a successful HTTP response.
    private final Pattern successStatusLine = Pattern.compile("^http/1\\.(0|1) (?<statusCode>2[0-9]{2})", Pattern.CASE_INSENSITIVE);
    private final Predicate<String> successStatusLinePredicate = successStatusLine.asPredicate();

    // Whatever the function is.
    boolean validateResponse() {
         final String firstLine = responseScanner.nextLine();
         if (successStatusLinePredicate.test(firstLine)) {
             // do stuff.
         }
    }
}
```

Reply:
works for me, when will this hit an official release so i can advice the project manager? this is a blocking issue for the go live of some features and we're in a bit of a hurry.

Reply:
created a new commit with the changes